hare/algorithm.go

```diff
@@ -281,9 +286,16 @@ PreRound:
 		case <-ticker.C: // next round event
 			proc.onRoundEnd()
 			proc.advanceToNextRound()
+
+			// exit if we reached the limit on number of iterations
+			if proc.k/4 > int32(proc.cfg.Limit) {
```

Comment:
should we advance to next round if we wish to terminate?

Reply:
Not important, but is the result of the division rounds down? if so, then this is not tight.
Also, I think that you can do =>  if the 'k' preround is a multiple of 4

Reply:
@antonlerner @barakshani  intuitively it is more correct to ask "should I terminate?" before I advance to the next round. On the other hand, I don't want to delay or make a more complicated query. So this point is not very important.

Rounds down. But what is the problem with it? The nodes are assumed to be synced on the same round. So, up to the drift, they will terminate.